src/error.rs

```diff
@@ -284,19 +276,330 @@ impl From<InvalidWindowHandle> for CmdError {
+            ScriptTimeout => "script timeout",
+            SessionNotCreated => "session not created",
+            StaleElementReference => "stale element reference",
+            Timeout => "timeout",
+            UnableToCaptureScreen => "unable to capture screen",
+            UnableToSetCookie => "unable to set cookie",
+            UnexpectedAlertOpen => "unexpected alert open",
+            UnknownCommand | UnknownError => "unknown error",
+            UnknownMethod => "unknown method",
+            UnknownPath => "unknown command",
+            UnsupportedOperation => "unsupported operation",
+        }
+    }
+
+    /// Returns the correct HTTP status code associated with the error type.
+    pub fn http_status(&self) -> StatusCode {
+        use self::ErrorStatus::*;
+        match *self {
+            DetachedShadowRoot => StatusCode::NOT_FOUND,
+            ElementClickIntercepted => StatusCode::BAD_REQUEST,
+            ElementNotInteractable => StatusCode::BAD_REQUEST,
+            ElementNotSelectable => StatusCode::BAD_REQUEST,
+            InsecureCertificate => StatusCode::BAD_REQUEST,
+            InvalidArgument => StatusCode::BAD_REQUEST,
+            InvalidCookieDomain => StatusCode::BAD_REQUEST,
+            InvalidCoordinates => StatusCode::BAD_REQUEST,
+            InvalidElementState => StatusCode::BAD_REQUEST,
+            InvalidSelector => StatusCode::BAD_REQUEST,
+            InvalidSessionId => StatusCode::NOT_FOUND,
+            JavascriptError => StatusCode::INTERNAL_SERVER_ERROR,
+            MoveTargetOutOfBounds => StatusCode::INTERNAL_SERVER_ERROR,
+            NoSuchAlert => StatusCode::NOT_FOUND,
+            NoSuchCookie => StatusCode::NOT_FOUND,
+            NoSuchElement => StatusCode::NOT_FOUND,
+            NoSuchFrame => StatusCode::NOT_FOUND,
+            NoSuchShadowRoot => StatusCode::NOT_FOUND,
+            NoSuchWindow => StatusCode::NOT_FOUND,
+            ScriptTimeout => StatusCode::INTERNAL_SERVER_ERROR,
+            SessionNotCreated => StatusCode::INTERNAL_SERVER_ERROR,
+            StaleElementReference => StatusCode::NOT_FOUND,
+            Timeout => StatusCode::INTERNAL_SERVER_ERROR,
+            UnableToCaptureScreen => StatusCode::BAD_REQUEST,
+            UnableToSetCookie => StatusCode::INTERNAL_SERVER_ERROR,
+            UnexpectedAlertOpen => StatusCode::INTERNAL_SERVER_ERROR,
+            UnknownCommand => StatusCode::NOT_FOUND,
+            UnknownError => StatusCode::INTERNAL_SERVER_ERROR,
+            UnknownMethod => StatusCode::METHOD_NOT_ALLOWED,
+            UnknownPath => StatusCode::NOT_FOUND,
+            UnsupportedOperation => StatusCode::INTERNAL_SERVER_ERROR,
+        }
+    }
+}
+
+impl Serialize for ErrorStatus {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        self.error_code().serialize(serializer)
+    }
+}
+
+impl From<webdriver::ErrorStatus> for ErrorStatus {
```

Comment:
Unfortunately this `impl` isn't okay, as it'd still mean we're tied to the major version of the `webdriver` crate. This will have to be a private inherent method on `ErrorStatus` instead.

Reply:
I agree with the suggestion but it turns out this isn't actually used anywhere. Should be fine to remove it I think.

Reply:
Removed